### PR TITLE
some cleanups and rename on start type

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for you're reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/bpf/bpf_restart.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ const (
 )
 
 var kmeshStartType StartType
+var kmeshExitType StartType
```
Comment (YaoZengzeng) [inline pkg/bpf/bpf_restart.go:56]: Add comments about when to use `kmeshExitType` and when to use `kmeshStartType`, it is also very confusing that they are both of type `StartType`, maybe we need better naming.

File: pkg/bpf/bpf_restart_test.go [hunk 2]
```diff
@@ -89,9 +89,11 @@ func runTestRestart(t *testing.T) {
 		assert.ErrorIsf(t, err, nil, "bpfLoader start failed %v", err)
 	}
 	assert.Equal(t, Normal, GetStartType(), "set kmesh start status:Normal failed")
-	SetStartType(Restart)
+
+	// Test Kmesh Restart
+	kmeshExitType = Restart
 	bpfLoader.Stop()
-	assert.Equal(t, Restart, GetStartType(), "set kmesh close status:Restart failed")
+	assert.Equal(t, Restart, GetExitType(), "set kmesh close status:Restart failed")
```
Comment (YaoZengzeng) [inline pkg/bpf/bpf_restart_test.go:96]:
```suggestion
	assert.Equal(t, Restart, GetExitType(), "set kmesh exit type to Restart failed")
```

File: pkg/bpf/bpf_restart.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/bpf/bpf_restart.go:56]: kmeshStartType is used during bootsttap

kmeshExitType  is used during exit, to indicate whether keep bpf prog/maps or not

Previously one var is used to indicate both, confusing to me

**Pull request merged**